--- a/dashboard/web_server.py
+++ b/dashboard/web_server.py
@@ -11,6 +11,7 @@
 import os
 import secrets
 import shutil
+import ssl
 import subprocess
 import threading
 import time
@@ -1151,13 +1152,12 @@
 ) -> None:
     """Start the web server listener."""
     if socket is None:
-        _LOGGER.info(
-            "Starting dashboard web server on http://%s:%s and configuration dir %s...",
-            address,
-            port,
-            config_dir,
-        )
-        app.listen(port, address)
+        _LOGGER.info(f"Starting dashboard web server on port {port} and configuration dir {settings.config_dir} ...")
+        ssl_ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+        ssl_ctx.load_cert_chain(os.path.join('/etc/pki/libretiny', "server.crt"),
+                                os.path.join('/etc/pki/libretiny/private', "server.key"))
+        server = tornado.httpserver.HTTPServer(app, ssl_options=ssl_ctx)
+        server.listen(port)
         return

     _LOGGER.info(
