--- a/esphome/dashboard/dashboard.py
+++ b/esphome/dashboard/dashboard.py
@@ -12,6 +12,7 @@
 import os
 import secrets
 import shutil
+import ssl
 import subprocess
 import threading
 from pathlib import Path
@@ -1345,13 +1346,13 @@
         socket = tornado.netutil.bind_unix_socket(args.socket, mode=0o666)
         server.add_socket(socket)
     else:
-        _LOGGER.info(
-            "Starting dashboard web server on http://%s:%s and configuration dir %s...",
-            args.address,
-            args.port,
-            settings.config_dir,
-        )
-        app.listen(args.port, args.address)
+        _LOGGER.info(f"Starting dashboard web server on port {args.port} and configuration dir {settings.config_dir} ...")
+        ssl_ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+        ssl_ctx.load_cert_chain(os.path.join('/etc/pki/libretiny', "server.crt"),
+                                os.path.join('/etc/pki/libretiny/private', "server.key"))
+
+        server = tornado.httpserver.HTTPServer(app, ssl_options=ssl_ctx)
+        server.listen(args.port)

         if args.open_ui:
             import webbrowser
